src/main/resources/assets/js/application/recruiter/directives/tdprSlotDirective.js

```diff
@@ -0,0 +1,18 @@
+define(['angular', 'application/recruiter/tdprRecruiterModule'], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.directive("slot", function () {
+        return {
+            restrict: 'AE',
+            templateUrl: 'js/application/recruiter/views/tdpr-directive-slot.html',
+            replace: true,
+            scope: {
+                slotData: '='
+
+            },
+            link: function (scope, element, attributes) {
+                scope.getClass = function () {
+                  return scope.slotData ? 'slot-' + scope.slotData.type : '';
```

Comment:
its safer to get it from the data we recieve from server. enum on front end is not needed